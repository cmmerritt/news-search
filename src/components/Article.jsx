import React from 'react';
import PropTypes from 'prop-types';

// image url = urlToImage
// grab everything from json.articles.[key] e.g. json.articles.title
// unique id: use url?

const Article = ({ title, description, author, url, image }) => (
  <figure>
    <img src={image} alt={title} />
    <figcaption>
      <p>{title}</p>
      <p>{description}</p>
      <p>{author}</p>
      <p>{url}</p>
    </figcaption>
  </figure>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  author: PropTypes.string,
  url: PropTypes.string.isRequired,
  image: PropTypes.string
};

export default Article;
