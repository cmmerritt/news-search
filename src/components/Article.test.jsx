/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  it('renders Article', () => {
    const { asFragment } = render(
      <Article 
        title="Test title"
        description="Test description"
        author="Test author"
        url="testurl.com"
        image="testimg.com"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
