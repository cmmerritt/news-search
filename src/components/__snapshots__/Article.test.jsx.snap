// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Test title"
      src="testimg.com"
    />
    <figcaption>
      <p>
        Test title
      </p>
      <p>
        Test description
      </p>
      <p>
        Test author
      </p>
      <p>
        testurl.com
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
